packages/eslint-plugin-pf-codemods/src/rules/helpers/renameSinglePropOnNode.ts

```diff
@@ -0,0 +1,94 @@
+import { JSXAttribute, JSXIdentifier, JSXOpeningElement } from 'estree-jsx';
+import { RenameConfig } from './renamePropsOnNode';
+import { Rule } from 'eslint';
+
+function isRenameConfig(obj: any): obj is RenameConfig {
+  const hasValidObjectType = typeof obj === 'object' && obj !== null;
+
+  const hasValidNewNameType = typeof obj.newName === 'string';
+
+  const isValidMessage =
+    obj.message === undefined ||
+    typeof obj.message === 'string' ||
+    typeof obj.message === 'function';
+
+  const isValidReplace =
+    obj.replace === undefined || typeof obj.replace === 'boolean';
+
+  return (
+    hasValidObjectType &&
+    hasValidNewNameType &&
+    isValidMessage &&
+    isValidReplace
+  );
+}
+
+const getNewName = (
+  replaced: boolean,
+  propRename: RenameConfig | string | Record<PropertyKey, never>
+) => {
+  if (!replaced) return '';
+  return (
+    isRenameConfig(propRename) ? propRename.newName : propRename
+  ) as string;
```

Comment:
Rather than having to assert here, wdyt about updating the if statement above to check whether `!replace || (![is not a rename config] && [is not a string])`, so that if propRename is an empty object we return the empty string as well?